docs/validator-committer.md

```diff
@@ -0,0 +1,332 @@
+    * [Task 3. Committing Valid Transactions to the Database](#task-3-committing-valid-transactions-to-the-database)
+6.  [gRPC Service API](#6-grpc-service-api)
+7.  [Failure and Recovery](#7-failure-and-recovery)
+
+## 1. Overview
+
+The [Validator-Committer (VC) service](https://github.com/hyperledger/fabric-x-committer/tree/main/service/vc) is a core component responsible for the final stages of transaction processing.
+It operates downstream from the Coordinator, receiving batches of conflict-free transactions. Its primary function
+is to perform optimistic concurrency control by validating each transaction's read-set against the current state in
+the database. Transactions that pass this validation are then committed, and their write-sets are applied to the database.
+
+## 2. Core Responsibilities
+
+The VC service performs three main tasks:
+
+1.  **Receive Transactions:** Receives batches of conflict-free transactions from the Coordinator via a gRPC stream.
+2.  **Validate and Commit:** Performs Multi-Version Concurrency Control (MVCC) checks on transaction read-sets against the committed
+database state. It commits valid transactions and reports the final status (e.g., committed, aborted due to MVCC conflict)
+of every transaction back to the Coordinator.
+3.  **Provide Data Access:** Exposes gRPC APIs that allow the Coordinator to read from and
+write to the state database, ensuring consistent data access.
+
+## 3. Configuration
+
+The VC service requires the following configuration settings, provided in a YAML [configuration file](https://github.com/hyperledger/fabric-x-committer/blob/main/cmd/config/samples/vcservice.yaml):
+
+* *Database Endpoints*: Address(es) of the state database nodes that the service will connect to.
+* *Listen Address*: The local address and port (e.g., 0.0.0.0:7056) where the VC service will host its gRPC service, enabling
+the Coordinator to connect and send transaction batches.
+
+## 4. Database Schema
+
+The VC service is responsible for managing several tables within the state database. These tables store the transaction statuses,
+world state for each namespace, and internal metadata. The tables are categorized into system tables and user namespace tables.
+
+### a. System Tables
+
+* **Transaction Status Table (`tx_status`)**: This table stores the final status of every transaction processed by the system.
+
+| Column Name | Data Type | Description/Constraints                                                 |
+| :---------- | :-------- | :---------------------------------------------------------------------- |
+| `tx_id`     | `BYTEA`   | The transaction ID. Primary Key.                                        |
+| `status`    | `INTEGER` | The final status code.                                                  |
+| `height`    | `BYTEA`   | A combination of the block number and the transaction's index within the block. |
```

Comment:
minor: is it worth mentioning that we use order preserving encoding?
nit: table not aligned